src/NSwag.CodeGeneration/Models/ParameterModelBase.cs

```diff
@@ -32,25 +32,69 @@ public abstract class ParameterModelBase
         protected ParameterModelBase(string parameterName, string variableName, string typeName,
             SwaggerParameter parameter, IList<SwaggerParameter> allParameters, CodeGeneratorSettingsBase settings, IClientGenerator generator)
         {
-            Type = typeName;
-            Name = parameterName;
-            VariableName = variableName;
-
             _allParameters = allParameters;
             _parameter = parameter;
             _settings = settings;
             _generator = generator;
+
+            Type = typeName;
+            Name = parameterName;
+            VariableName = variableName;
+
+            if ( !parameter.IsRequired && parameter.Default != null ) {
+                Default = DefaultToString( parameter );
+
+                // If a default value is supplied, then we can automatically set 
+                // it when using the controller interface indirection
+                TypeInControllerInterface = 

```

Comment:
Not sure this is the best name, its the type as used in the controller interface. Meaning that it strips the ? when there is a default.